### PR TITLE
Implement infection process

### LINKED ISSUE #7: We need an infection process
> Define our infection process (S -> E1)

See #2 for details

We should test that:

 * this process moves the correct individuals for a mocked return value of `bernoulli_multi_p`
 * the correct probabilities are passed into `bernoulli_multi_p` for a variety of ages

### PR DESCRIPTION
**Pull request by giovannic**
Closes #7 

Creates an infection process moving individuals from S to E.

Waiting on #32 to integrate into the model

### CODE REVIEW

File: R/infection_process.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      # Calculate FoI and use to create probability for each age group
+      m <- get_contact_matrix(pars$contact_matrix_set)
+      lambda <- pars$beta * rowSums(m %*% diag(inf_ages))
+
+      # Transition from S to E
+      susceptible <- api$get_state(human, states$S)
+      ages <- api$get_variable(human, discrete_age, susceptible)
+
+      # FOI for each susceptible person
+      lambda <- lambda[as.integer(ages)]
```
Comment (pwinskill) [inline R/infection_process.R]: If we are request `discrete_age` in line 29, is this `as.integer()` call needed?

File: R/parameters.R [hunk 2]
```diff
@@ -24,5 +24,7 @@ get_parameters <- function(pop, dt = 1, time_period = 365, ...) {
 #'
 #' @param countryname name of country
 get_population <- function(countryname) {
-  squire::get_population(countryname, simple_SEIR = FALSE)
+  pop <- squire::get_population(countryname, simple_SEIR = FALSE)
```
Comment (pwinskill) [inline R/parameters.R]: Don't think this is too important but are iso3c codes less ambigious than countrynames?

File: R/variables.R [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  # use these to work out the ages in each bin
+  ages_in_bin <- list()
+  for (i in seq_along(age_bins)[-1]) {
+    ages_in_bin[[i - 1]] <- seq(age_bins[i - 1], age_bins[i] - 1, 1)
+  }
+  ages_in_bin[[length(ages_in_bin) + 1]] <- seq(max(age_bins), max_age, 1)
+
+  # now sample from these
+  ages <- c()
+  for (i in seq_len(length(pop$age_group))) {
```
Comment (pwinskill) [inline R/variables.R]: Would `for (i in seq_along(pop$age_group)){` be more succinct here?

File: data-raw/prepare_raw_data.R [hunk 4]
```diff
@@ -0,0 +1,38 @@
+library(tidyverse)
+library(xml2)
+library(rvest)
+
+# 1. Median Country Ages -------------------------------------------------------
+
+# Decide if we trust worldometers ages
```
Comment (pwinskill) [inline data-raw/prepare_raw_data.R]: Just highlighting this - we probably don't want to push a comment like this into working code!

**Review by pwinskill — COMMENTED** [review]
Looks good.  A few minor comments here 😎

File: R/parameters.R (hunk 2 above)
Comment (OJWatson) [reply R/parameters.R]: I would be all over make the general switch to iso3c codes throughout. or either/or as we have in some of the squire functions.

**Comment by giovannic** [comment]
Made the changes. By the way, you have to click "approve" on your review to allow merging on this repo ;)

**Review by OJWatson — APPROVED** [review]

**Review by pwinskill — APPROVED** [review]

**Pull request merged**